### ISSUE TITLE
Crash on stopping world

### ISSUE THREAD

**Issue by ghost** [cuberite/cuberite#2286]
This plugin will teleport the player from one world to another world. I think it's the same problem like in #2056, but this occurs on stopping the world.

Requirements:
Add a world named test and the lua test plugin below.

Reproduce:
1) Connect to the server
2) Run the command test
3) After you have been teleported one ore more times, disconnect
4) Connect again to the server
5) Disconnect
6) Stop the server, it should crash

Lua test plugin:
```lua
function Initialize(Plugin)
    Plugin:SetName("TestPlugin")
    Plugin:SetVersion(1)
       
    -- Command Bindings
    cPluginManager.BindCommand("/test", "test", CmdTest , " test")
    
    LOG("Initialised " .. Plugin:GetName() .. " v." .. Plugin:GetVersion())
    return true
end

function CmdTest(a_Split, a_Player)
	local playerName = a_Player:GetName()
	CALLBACK = 
		function(a_World)
			a_World:DoWithPlayer(playerName,
				function(a_Player2)
					if (a_Player2:GetWorld():GetName() == "world") then
						a_Player2:MoveToWorld("test")
					else
						a_Player2:MoveToWorld("world")
					end
					a_Player2:GetWorld():ScheduleTask(20, CALLBACK)
				end)
		end
	a_Player:GetWorld():ScheduleTask(20, CALLBACK)
	return true
end
```

**Comment by madmaxoft** [comment]
So many crashes lately...

**Comment by Seadragon91** [comment]
Okay no need for the script.

1) Connect to the server
2) Teleport to a other world
3) Disconnect
4) Connect
5) Disconnect
6) Stop the server, it should crash

**Comment by worktycho** [comment]
It's the player destruction code. Does anyone fully understand how it works?

**Comment by Seadragon91** [comment]
I think this error counts to that issue, too.

```
#0  0x00000000005214da in cChunk::AddClient(cClientHandle*) ()
#1  0x000000000052a9bf in cChunkMap::AddChunkClient(int, int, cClientHandle*)
    ()
#2  0x0000000000534802 in cClientHandle::StreamChunk(int, int, cChunkSender::eChunkPriority) ()
#3  0x000000000053471b in cClientHandle::StreamNextChunk() ()
#4  0x000000000053891d in cClientHandle::Tick(float) ()
#5  0x00000000005a2eeb in cWorld::TickClients(float) ()
#6  0x00000000005946d0 in cWorld::Tick(std::chrono::duration<long, std::ratio<1l, 1000l> >, std::chrono::duration<long, std::ratio<1l, 1000l> >) ()
#7  0x00000000005943cc in cWorld::cTickThread::Execute() ()
#8  0x00007ffff71e5e40 in ?? () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#9  0x00007ffff74b2182 in start_thread ()
   from /lib/x86_64-linux-gnu/libpthread.so.0
#10 0x00007ffff6c4c47d in clone () from /lib/x86_64-linux-gnu/libc.so.6
```

**Comment by Seadragon91** [comment]
Any updates on this?

**Comment by LogicParrot** [comment]
@SphinxC0re No. Some "important" issues can be introduced in Master, but are not a problem at all in testing because that code is older.

**Comment by LogicParrot** [comment]
#2940 prevents the crash in this scenario:
```
1) Connect to the server
2) Run the command test
3) After you have been teleported one ore more times, disconnect
4) Connect again to the server
5) Disconnect
6) Stop the server, it should crash
```

**Comment by SphinxC0re** [comment]
Then just merge it as usual

**Comment by LogicParrot** [comment]
@SphinxC0re you posted in the wrong issue :)

**Comment by LogicParrot** [comment]
Do #2939 and/or #2941 fix this?

**Comment by SphinxC0re** [comment]
I don't really know. Could you file a test case?

**Comment by LogicParrot** [comment]
A test case would be running @ghost's initial instructions over the current master.

**Comment by LogicParrot** [comment]
Fixed in #2941
